src/main/java/com/google/devtools/build/lib/rules/java/JavaOptions.java

```diff
@@ -597,11 +610,22 @@ public ImportDepsCheckingLevelConverter() {
               + "--java_header_compilation is enabled.")
   public boolean requireJavaToolchainHeaderCompilerDirect;
 
+  private Label getHostJavaBase() {
+    if (hostJavaBase == null) {
+      if (useRemoteJdkAsHostJavaBase && "1".equals(System.getProperty("embedded_jdk"))) {
```

Comment:
Can you document this part? Why would we only use the remote JDK with binaries that have an embedded JDK?

Reply:
Added a comment - we do want to make the switch as soft as possible.